### PR TITLE
User story 13

### PR DESCRIPTION
**Pull request by jhgould**
adds login ability for regular users, admins, and merchants.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        session[:user_id] = user.id
+        if user.role == "admin" 
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/admin/dashboard"
+        elsif user.role == "merchant"
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/merchant/dashboard"
+        else 
+            flash[:notice] = "You are logged in as #{user.name}"
+            redirect_to "/profile"
```
Comment (jhgould) [inline app/controllers/sessions_controller.rb:17]: Based on their role, user gets redirected to a different route. Admin and Merchant routes are namespaced.

File: app/controllers/merchant_dashboard_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class MerchantDashboardController < ApplicationController
+    def show 
+        binding.pry
+    end
```
Comment (jhgould) [inline app/controllers/merchant_dashboard_controller.rb:4]: I think we can delete this file.

File: db/migrate/20200527194008_add_role_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :role, :integer, default: 0
```
Comment (whitneykidd) [inline db/migrate/20200527194008_add_role_to_users.rb:3]: I'm glad we added this as early on as we did 🥇

File: app/controllers/merchant_dashboard_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class MerchantDashboardController < ApplicationController
+    def show
```
Comment (EricLarson2020) [inline app/controllers/merchant_dashboard_controller.rb:2]: consider whether we should continue with merchant dashboard controller

File: app/controllers/merchant_dashboard_controller.rb (hunk 2 above)
Comment (EricLarson2020) [reply app/controllers/merchant_dashboard_controller.rb:4]: You might be right, Ill take a look as we do the next few user stories

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (EricLarson2020) [reply app/controllers/sessions_controller.rb:17]: ya, not sure if we want to name space the merchant, I think it works as is though, but I will play around with it for a bit

File: spec/features/login/login_spec.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        
+   jack = User.create!({
+      name: "Jack",
+      address: "333 Jack Blvd",
+      city: "Denver",
+      state: "Colorado",
+      zip: 83243,
+      email: "jack@hotmail.com",
+      password: "3455", 
+      password_confirmation: "3455",
```
Comment (EricLarson2020) [inline spec/features/login/login_spec.rb:15]: looks good

**Pull request merged**